velox/parse/DuckLogicalOperator.h

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+// This file contains the definitions of the logical plan nodes extracted from
+// the duckdb-internal.hpp file. It is not possible to include
+// duckdb-internal.hpp directly because it contains an incompatible / outdated
+// copy of fmt/format.h.
+
+/*
+Copyright 2018-2022 Stichting DuckDB Foundation
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of
+this software and associated documentation files (the "Software"), to deal in
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+#include "velox/external/duckdb/duckdb.hpp"
+
+namespace duckdb {
+
+//! LogicalDummyScan represents a dummy scan returning a single row
+class LogicalDummyScan : public LogicalOperator {
+ public:
+  explicit LogicalDummyScan(idx_t table_index)
+      : LogicalOperator(LogicalOperatorType::LOGICAL_DUMMY_SCAN),
+        table_index(table_index) {}
+
+  idx_t table_index;
+
+ public:
+  vector<ColumnBinding> GetColumnBindings() override {
+    return {ColumnBinding(table_index, 0)};
+  }
+
+  idx_t EstimateCardinality(ClientContext& context) override {
+    return 1;
+  }
+
+ protected:
+  void ResolveTypes() override {
+    if (types.size() == 0) {
```

Comment:
Please can you add a comment to explain this behavior.

Reply:
This is a copy-paste from duckdb-internal.hpp. Hence, cannot modify.